### ISSUE TITLE
Poller deactivation race condition.

### ISSUE THREAD

**Issue by tardyp** [buildbot/buildbot#3217]
<kevin``> hey buddy, my gitpoller stop working! and I use buildbot 0.9.5, the issue log has been pasted on https://pastebin.com/4rkfRXXE
```
2017-05-16 23:00:03+0800 [-] while polling for changes
        Traceback (most recent call last):
          File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 1184, in gotResult
            _inlineCallbacks(r, g, deferred)
          File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 1128, in _inlineCallbacks
            result = g.send(result)
          File "/usr/local/lib/python2.7/dist-packages/buildbot/changes/gitpoller.py", line 208, in poll
            yield self.setState('lastRev', self.lastRev)
          File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 1274, in unwindGenerator
            return _inlineCallbacks(None, gen, Deferred())
        --- <exception caught here> ---
          File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 1128, in _inlineCallbacks
            result = g.send(result)
          File "/usr/local/lib/python2.7/dist-packages/buildbot/util/state.py", line 45, in setState
            yield self.master.db.state.setState(self._objectid, key, value)
        exceptions.AttributeError: 'NoneType' object has no attribute 'db'
 
2017-05-16 23:00:03+0800 [-] Caught exception while deactivating ClusteredService(git@192.168.11.2:bb/aaa.git)
        Traceback (most recent call last):
          File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 393, in callback
            self._startRunCallbacks(result)
          File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 501, in _startRunCallbacks
            self._runCallbacks()
          File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 588, in _runCallbacks
            current.result = callback(current.result, *args, **kw)
          File "/usr/local/lib/python2.7/dist-packages/buildbot/util/service.py", line 319, in <lambda>
            lambda _: defer.maybeDeferred(self._unclaimService))
        --- <exception caught here> ---
          File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 150, in maybeDeferred
            result = f(*args, **kw)
          File "/usr/local/lib/python2.7/dist-packages/buildbot/changes/base.py", line 53, in _unclaimService
            return self.master.data.updates.trySetChangeSourceMaster(self.serviceid,
        exceptions.AttributeError: 'NoneType' object has no attribute 'data'
 
2017-05-16 23:00:03+0800 [-] configuration update complete
```
03:55:22 I<infobob> https://paste.pound-python.org/show/puucNt0lxjmsIcRQ5M5G/ (repasted for kevin``)
03:56:49 K<kevin``> I wonder, if my `buildbot reconfig master' operation leads to this issue, any help? thanks
03:59:51 S<+skelly> try a restart
04:03:48 K<kevin``> restart can fix it.
04:04:10 K<kevin``> I think I find the root cause.
04:04:40 K<kevin``> last gitpoller log: 2017-05-16 22:55:03+0800 [-] gitpoller: processing changes from
04:05:20 K<kevin``> then 2017-05-16 23:00:02+0800 [-] beginning configuration update
04:05:39 K<kevin``> 2017-05-16 23:00:02+0800 [-] Loading configuration from
04:05:50 K<kevin``> 2017-05-16 23:00:03+0800 [-] gitpoller: processing changes from 
04:06:03 K<kevin``> my poller interval is 5 minutes
04:06:46 K<kevin``> So, the polling action and the loading configuration action happens at the same time
04:07:00 K<kevin``> Then the exception happens.

**Comment by ciarancourtney** [comment]
I just got this error in 1.5.0

**Comment by p12tic** [comment]
I believe this has been fixed in #5798. The parent of the service was being cleared before the service itself was shut down so master instance was no longer accessible and any actions that depend on the master would fail during that time period.
